tests/test_set_config.py

```diff
@@ -492,3 +494,41 @@ def test_erroneous_conversion_flags(): #pylint: disable=unused-variable
 
         with pytest.raises(ValueError):
             set_config.create_config_obj(args)
+
+def test_compare_nml(): #pylint: disable=unused-variable
+
+    ''' Tests whether comparing two namelists works. '''
+
+    nml1 = os.path.join(uwtools_file_base, pathlib.Path("fixtures/fruit_config.nml"))
+    nml2 = os.path.join(uwtools_file_base,
+                        pathlib.Path("fixtures/fruit_config_mult_sect.nml"))
+
+    args = ['-i', nml1, '-c', nml2, '--compare']
+
+    expected = f"""
+    - {nml1}
+    + {nml2}
+    --------------------------------------------------------------------------------
+    config:       vegetable:  - eggplant + peas
+    setting:         topping:  - None + crouton
+    setting:            size:  - None + large
+    setting:            meat:  - None + chicken
+
+    """
+
+    outstring = io.StringIO()
+    with redirect_stdout(outstring):
+        set_config.create_config_obj(args)
+    result = outstring.getvalue()
+
+    # Make sure the tool output contains all the lines above
+    for expected_line in expected.split('\n'):
+        assert expected_line.strip() in result
+
+    # Make sure it doesn't include any additional significant diffs
+    # A very rough estimate is that there is a word/colon set followed
+    # by a -/+ set
+    pattern = re.compile(r'(\w):\s+(\w+):\s+-\s+(\w+)\s+\+\s+(\w+)')
```

Comment:
Oof, no way to avoid regex?

Reply:
I thought I was the only one who didn't like regex :). If it can't easily be done another way, maybe an example or two to make it easier to decode?

Reply:
@christinaholtNOAA What was the error or issue that needed a second assert with regex?

Reply:
Thanks for sorting us out in the stand-up, there is a time and place for regex and validating logs is one that we've been avoiding but needed eventually

Reply:
I added a comment here that is hopefully a bit clearer.